cmd/restic/cmd_stats.go

```diff
@@ -101,7 +103,7 @@ func runStats(gopts GlobalOptions, args []string) error {
 
 		var sID restic.ID
 		if snapshotIDString == "latest" {
```

Comment:
Instead of this whole determination of the snapshot (if given) by `snapshotIDString := arg[0]`, IMO we should use
```golang
	for sn := range FindFilteredSnapshots(ctx, repo, opts.Hosts, opts.Tags, opts.Paths, args) {
		err = statsWalkSnapshot(ctx, sn, repo, stats)
		if err != nil {
			return fmt.Errorf("error walking snapshot: %v", err)
		}
	}
```

This would make `stats` behave like forget w.r.t. the given command line flags and arguments and solve the issue of @faizshukri while also making the code much cleaner. Moreover, the actual behavior (i.e. the only the two supported cases "no arguments = all snapshots" and "latest") would work almost identically. The only thing to change would be that the flag `--host` without `latest` would now not ignore the `--host` anymore.

However this change would also need some small additional modifications where `snapshotIDString` is actually used. 
@buschjost Are you ok to change this or should I propose another PR to implement this?

Reply:
@aawsome I was not sure about the reason of the current implementation and wanted to avoid breaking something. 

I had a short look on your suggestion and it feels like a change cleaning up many things. FindFilteredSnapshots seems to handle all cases. 

I'll have a look at adjusting it.

Reply:
As specifying the `--host` option without `latest` should have thrown an error in the first place it should be fine to change this. After all I'd expected that the `--host` option is taken into account when passed to the `stats` command.

The error handling of `FindFilteredSnapshots` is different from the current behavior: Currently the command just aborts if an error occurs. FindFilteredSnapshots on the other hand just prints a warning message for the user and continues. This could become a problem when using the JSON interfaces. But in the end it just exchanges one invalid JSON output with another one, so it should be fine I guess.